scalapact-core/src/main/scala/com/itv/scalapactcore/common/matching/InteractionMatchingPrograms.scala

```diff
@@ -0,0 +1,213 @@
+package com.itv.scalapactcore.common.matching
+
+import com.itv.scalapactcore.common.matching.PathMatching.PathAndQuery
+
+import scalaz._
+import Scalaz._
+import com.itv.scalapactcore.{InteractionRequest, InteractionResponse}
+
+sealed trait MatchStatusOutcome
+case object MatchStatusSuccess extends MatchStatusOutcome
+case object MatchStatusFailed extends MatchStatusOutcome
+
+sealed trait MatchHeadersOutcome
+case object MatchHeadersSuccess extends MatchHeadersOutcome
+case object MatchHeadersFailed extends MatchHeadersOutcome //TODO: What kind of match failures do we get that we can model?
+
+sealed trait MatchBodyOutcome
+case object MatchBodySuccess extends MatchBodyOutcome
+case object MatchBodyFailed extends MatchBodyOutcome //TODO: What kind of match failures do we get that we can model?
+
+sealed trait MatchMethodOutcome
+case object MatchMethodSuccess extends MatchMethodOutcome
+case object MatchMethodFailed extends MatchMethodOutcome //TODO: What kind of match failures do we get that we can model?
+
+sealed trait MatchPathOutcome
+case object MatchPathSuccess extends MatchPathOutcome
+case object MatchPathFailed extends MatchPathOutcome //TODO: What kind of match failures do we get that we can model?
```

Comment:
Lot's of noise here. None of this (above) has anything to do with Free Monads. It's just me wanting - one day - to return something more meaningful that a Boolean!